### ISSUE TITLE
Changing namespace doesn't preserve enum defaults

### ISSUE THREAD

**Issue by bcarter97** [sksamuel/avro4s#690]
Hi, 

when overriding a namespace of an enum in a case class I am not seeing the enum default in the generated schema.

For example, the following class `Bowl`

```scala
@AvroEnumDefault(Unknown)
@AvroNamespace("com.default.package")
sealed trait Fruit

@AvroUnionPosition(0) case object Unknown extends Fruit
@AvroUnionPosition(1) case object Orange  extends Fruit
@AvroUnionPosition(2) case object Mango   extends Fruit

@AvroNamespace("com.fruit")
case class Bowl(fruit: Fruit = Mango)

AvroSchema[Bowl]
```

Generates the schema as expected:

```json
{
  "type": "record",
  "name": "Bowl",
  "namespace": "com.fruit",
  "fields": [
    {
      "name": "fruit",
      "type": {
        "type": "enum",
        "name": "Fruit",
        "namespace": "com.default.package",
        "symbols": [
          "Unknown",
          "Orange",
          "Mango"
        ],
        "default": "Unknown"
      },
      "default": "Mango"
    }
  ]
}
```

However, overriding the namespace of the fruit as so:

```diff
- case class Bowl(fruit: Fruit = Mango)
+ case class Bowl(@AvroNamespace("com.fruit.bowl") fruit: Fruit = Mango)
```

Doesn't include the enum default in the generated schema:

```diff
{
  "type": "record",
  "name": "Bowl",
  "namespace": "com.fruit",
  "fields": [
    {
      "name": "fruit",
      "type": {
        "type": "enum",
        "name": "Fruit",
-      "namespace": "com.default.package",
+      "namespace": "com.fruit.bowl",
        "symbols": [
          "Unknown",
          "Orange",
          "Mango"
        ]
-       "default": "Unknown"
      },
      "default": "Mango"
    }
  ]
}
```

Is this intended behaviour, and if so, is there a way to create enum defaults on case classes with overrided namespaces?

**Comment by sksamuel** [comment]
Which version of avro4s / scala ?

**Comment by bcarter97** [comment]
Scala `2.12.11`, avro4s `4.0.11`

**Comment by sksamuel** [comment]
Fixed in 4.0.12.

**Comment by bcarter97** [comment]
Thank you 😄
